### PR TITLE
Fix/outdent source refs

### PR DESCRIPTION
**Pull request by neoxelox**
Fixes source map references drift when trimming and outdenting.

### CODE REVIEW

File: src/compiler/base/types.ts [hunk 1]
```diff
@@ -63,7 +63,7 @@ export type CompileNodeContext<N extends TemplateNode> = {
 
   setConfig: (config: Config) => void
   addMessage: (message: Message, global?: boolean) => void
-  addStrayText: (text: string, node: TemplateNode) => void
+  addStrayText: (text: string, node?: TemplateNode) => void
```
Comment (csansoon) [inline src/compiler/base/types.ts:66]: nicee

File: src/compiler/compile.test.ts [hunk 2]
```diff
@@ -698,5 +698,150 @@ Given a context, answer questions succintly yet complete.
...
+              text: "Please, help me\n  with question!",
+              _promptlSourceMap: [
+                {
+                  start: 23,
+                  end: 31,
+                  identifier: "question",
+                },
+              ],
+            },
+          ],
```
Comment (csansoon) [inline src/compiler/compile.test.ts:842]: Looks good 🙌

File: src/compiler/compile.ts [hunk 3]
```diff
@@ -184,15 +185,55 @@ export class Compile {
...
+    return sourceMap
+  }
+
   private groupStrayText(): void {
-    const text = this.popStrayText()
-    if (!text.text.trim().length) return
+    const stray = this.popStrayText()
+    const text = removeCommonIndent(stray.text)
+    const sourceMap = this.outdentSourceRefs(stray);
+
```
Comment (csansoon) [inline src/compiler/compile.ts:229]: Removing common indent from `stray.text` could not mess with the contents of `stray.text` before you run `outdentSourceRefs`?

A less hacky way would be to both perform the indent removal and source refs recalculation in the same method (?)

Comment (csansoon) [reply src/compiler/compile.ts:229]: If it works, it works though

Comment (neoxelox) [reply src/compiler/compile.ts:229]: Good catch, it could be, but I think strings are immutable in javascript, I will change the order tho!!

**Review by csansoon — APPROVED** [review]
Check minor comments.

Then, `pnpm version patch` and palante

File: src/compiler/compile.test.ts [hunk 4]
```diff
@@ -698,5 +698,150 @@ Given a context, answer questions succintly yet complete.
...
+
+
+  </system>
+      <user>
+
+  Please, help me
+    with {{ question }}!
+  </user>
+
+    {{empty}}{{empty}}
```
[the remark this answers is not in the document]
Comment (neoxelox) [reply src/compiler/compile.test.ts]: Good catch! There was a bug!

File: package.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@latitude-data/promptl",
-  "version": "0.3.0",
+  "version": "0.3.1",
```
Comment (csansoon) [inline package.json:3]: After rebasing with main it will be 0.3.2

File: src/compiler/compile.test.ts [hunk 6]
```diff
@@ -522,9 +522,9 @@ Given a context, answer questions succintly yet complete.
   describe("includes source map when specified", async () => {
     it("returns empty source map when no identifiers", async () => {
       const prompt = `
-  Given a context, answer questions succintly yet complete.
-  <system>context</system>
-  <user>Please, help me with question!</user>
+Given a context, answer questions succintly yet complete.
+<system>context</system>
+<user>Please, help me with question!</user>
```
Comment (csansoon) [inline src/compiler/compile.test.ts:527]: Why did you remove the indent here?

Comment (neoxelox) [reply src/compiler/compile.test.ts:527]: TOC and consistency with other tests. Indent thingies are tested in the specific test.

File: src/compiler/base/nodes/tags/content.ts [hunk 7]
```diff
@@ -54,13 +53,16 @@ export async function compile(
...
     addContent({
       node,
       content: {
         ...attributes,
         type: ContentType.text,
-        text: textContent,
+        text: stray.text,
+        _promptlSourceMap: stray.sourceMap,
       },
     })
```
Comment (csansoon) [inline src/compiler/base/nodes/tags/content.ts:67]: Add the `includeSourceMap` as a prop in the `CompileNodeContext`, and include it conditionally like you're doing in `compile`.

File: package.json (hunk 5 above)
Comment (neoxelox) [reply package.json:3]: Did you publish the new version yet? We can release both changes in the same version

File: src/compiler/base/nodes/tags/content.ts (hunk 7 above)
Comment (csansoon) [reply src/compiler/base/nodes/tags/content.ts:67]: https://github.com/latitude-dev/promptl/blob/c72c963a84b30a0e378ad637b0e1cc9df409686f/src/compiler/base/types.ts#L46

File: package.json (hunk 5 above)
Comment (csansoon) [reply package.json:3]: yep i did

**Review by csansoon — APPROVED** [review]
